uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -11,35 +11,32 @@ define([
   'leave-absences/shared/models/entitlement-model',
   'leave-absences/shared/models/leave-request-model',
   'leave-absences/shared/models/public-holiday-model',
-  'leave-absences/shared/models/instances/leave-request-instance',
-  'leave-absences/shared/models/instances/sickness-leave-request-instance',
-], function (components, _, moment) {
+], function (controllers, _, moment) {
   'use strict';
 
-  components.controller('LeaveRequestPopupCtrl', [
-    '$log', '$q', '$rootScope', '$uibModalInstance', 'Contact', 'AbsencePeriod', 'AbsenceType',
-    'api.optionGroup', 'directiveOptions', 'Calendar', 'Entitlement', 'HR_settings',
-    'LeaveRequest', 'LeaveRequestInstance', 'PublicHoliday', 'SicknessRequestInstance', 'shared-settings',
-    function ($log, $q, $rootScope, $modalInstance, Contact, AbsencePeriod, AbsenceType,
-      OptionGroup, directiveOptions, Calendar, Entitlement, HR_settings,
-      LeaveRequest, LeaveRequestInstance, PublicHoliday, SicknessRequestInstance, sharedSettings
+  controllers.controller('RequestCtrl', [
+    '$log', '$q', '$rootScope', 'Contact', 'AbsencePeriod', 'AbsenceType',
+    'api.optionGroup', 'Calendar', 'Entitlement', 'HR_settings',
+    'LeaveRequest', 'PublicHoliday', 'shared-settings',
+    function ($log, $q, $rootScope, Contact, AbsencePeriod, AbsenceType,
+      OptionGroup, Calendar, Entitlement, HR_settings,
+      LeaveRequest, PublicHoliday, sharedSettings
     ) {
-      $log.debug('LeaveRequestPopupCtrl');
+      $log.debug('RequestCtrl');
 
       var absenceTypesAndIds,
         initialLeaveRequestAttributes = {}, //used to compare the change in leaverequest in edit mode
         mode = '', //can be edit, create, view
         role = '', //could be manager, owner or admin
         selectedAbsenceType = {},
-        leaveType = 'leave', //other values could be sick or toil
         vm = {};
 
       vm.absencePeriods = [];
       vm.absenceTypes = [];
       vm.calendar = {};
       vm.contact = {};
       vm.error = null;
-      vm.leaveRequestDayTypes = [];
+      vm.requestDayTypes = [];
```

Comment:
1. Here `vm` is not local/parent because `vm` is returned at the end of the function. So all the properties assigned to `vm` is passed on to the child. Hence `this` should be used to keep consistency.

2. If some of the variables are not to be shared with child they should declared as private variables like `mode`.